app/controllers/providers/correspondence_address/choices_controller.rb

```diff
@@ -10,6 +10,12 @@ def show
       def update
         @form = Addresses::ChoiceForm.new(form_params)
 
+        delete_no_fixed_residence if form_params[:correspondence_address_choice].eql?("home")
```

Comment:
Does it follow our standard pattern to do this update in the form class?

Reply:
Updated in final commit